flink-runtime/src/main/java/org/apache/flink/runtime/state/memory/MemoryBackendCheckpointStorageAccess.java

```diff
@@ -72,6 +73,7 @@ public MemoryBackendCheckpointStorageAccess(
             JobID jobId,
             @Nullable Path checkpointsBaseDirectory,
             @Nullable Path defaultSavepointLocation,
+            boolean createCheckpointSubDirs,
```

Comment:
How about introducing another constructor to make `createCheckpointSubDirs` as default `true`? If so, we can avoid touching many test code in this PR.

Reply:
I'd rather keep a single constructor and not providing such constructor, since it is more friendly for further developing. WDYT?

Reply:
I have no strong against here. We can keep the current design.